src/faction.cpp

```diff
@@ -157,17 +153,61 @@ std::string faction::describe() const
     return ret;
 }
 
+void faction_power_spec::deserialize( const JsonObject &jo )
+{
+    mandatory( jo, false, "faction", faction );
+    optional( jo, false, "power_min", power_min );
+    optional( jo, false, "power_max", power_max );
+
+    if( !power_min.has_value() && !power_max.has_value() ) {
+        jo.throw_error( "Must have either a power_min or a power_max" );
+    }
+}
+
+void faction_epilogue_data::deserialize( const JsonObject &jo )
+{
+    optional( jo, false, "power_min", power_min );
+    optional( jo, false, "power_max", power_max );
+    optional( jo, false, "dynamic", dynamic_conditions );
+    mandatory( jo, false, "id", epilogue );
+}
+
+
+bool faction::check_relations( std::vector<faction_power_spec> faction_power_specs ) const
+{
+    if( !faction_power_specs.empty() ) {
+        for( const faction_power_spec &spec : faction_power_specs ) {
+            if( spec.power_min.has_value() ) {
+                if( spec.faction->power < spec.power_min.value() ) {
+                    return false;
+                }
+            }
+            if( spec.power_max.has_value() ) {
+                if( spec.faction->power >= spec.power_max.value() ) {
+                    return false;
+                }
+            }
+        }
+    }
+    return true;
```

Comment:
You don't need to do this, but it can be nice to cut down on indents by doing an early exit:
```suggestion
    if( faction_power_specs.empty() ) {
        return true;
    }
    for( const faction_power_spec &spec : faction_power_specs ) {
        if( spec.power_min.has_value() ) {
            if( spec.faction->power < spec.power_min.value() ) {
                return false;
            }
        }
        if( spec.power_max.has_value() ) {
            if( spec.faction->power >= spec.power_max.value() ) {
                return false;
            }
        }
    }
    return true;
```